cmake/faceliftMacros.cmake

```diff
@@ -402,13 +419,43 @@ endmacro()
 
 macro(_facelift_add_target_start IMPLEMENTATION_TARGET_NAME)
 
-    set(HEADERS_TO_BE_MOCCED ${HEADERS} ${HEADERS_NO_INSTALL})
+    set(HEADERS_FOR_MOC ${HEADERS_GLOB} ${HEADERS_NO_INSTALL})
+
+    set(MOC_MACRO_NAMES "Q_OBJECT" "Q_GADGET" "Q_NAMESPACE")
+
+    # check if all the autogenerated files contain specified macros (MOC_MACRO_NAMES)
+    foreach(HEADER ${HEADERS_FOR_MOC})
+        file(READ ${HEADER} FILE_STRING)
+     
+        unset(IS_FOUND)
+        foreach(MACRO_NAME ${MOC_MACRO_NAMES})
+            string(FIND "${FILE_STRING}" "${MACRO_NAME}" IS_FOUND)
```

Comment:
You are performing that grep every time the project is reconfigured.
The code generator should generate the list of MOCed files once to avoid that.

Reply:
All the files have been already generated and we grep them.

Isn't this the same if we will do it here instead of grep a file by Generator after a template processing ?

Reply:
The code generator is only triggered at configure time if really needed. There is no reason to perform a grep if the code generator is not triggered.